src/show.jl

```diff
@@ -67,13 +67,13 @@ function show_oneline(io::IO, d::Distribution, namevals)
     print(io, ')')
 end
 
-function show_multline(io::IO, d::Distribution, namevals)
+function show_multline(io::IO, d::Distribution, namevals; newline=true)
     print(io, distrname(d))
     println(io, "(")
     for (p, pv) in namevals
         print(io, p)
         print(io, ": ")
         println(io, pv)
     end
-    println(io, ")")
+    newline ? println(io, ")") : print(io, ")")
```

Comment:
IIRC there was a PR a while ago in which it was concluded the verbose display should always end with a newline character. What's the motivation for this change?

Reply:
Oh really? I wasn't aware of that. I always felt this was a mistake in Distributions. It means that types with multiline show always display with 2 newlines in the REPL instead of 1, where I can't think of any types in the standard library that do that, and I've probably only encountered 1 or 2 other types across the ecosystem that do.

But in this case, you can see the difference by comparing the results for `Truncated` vs `Censored`:
```julia
julia> censored(MixtureModel(fill(Normal(), 2)), -1, 1)
Censored(MixtureModel{Univariate, Continuous, Normal{Float64}, Categorical{Float64, Vector{Float64}}}(
components: Normal{Float64}[Normal{Float64}(μ=0.0, σ=1.0), Normal{Float64}(μ=0.0, σ=1.0)]
prior: Categorical{Float64, Vector{Float64}}(support=Base.OneTo(2), p=[0.5, 0.5])
), range=(-1, 1))

julia> truncated(MixtureModel(fill(Normal(), 2)), -1, 1)
Truncated(MixtureModel{Univariate, Continuous, Normal{Float64}, Categorical{Float64, Vector{Float64}}}(
components: Normal{Float64}[Normal{Float64}(μ=0.0, σ=1.0), Normal{Float64}(μ=0.0, σ=1.0)]
prior: Categorical{Float64, Vector{Float64}}(support=Base.OneTo(2), p=[0.5, 0.5])
)
, range=(-1.0, 1.0))


```
For `Truncated`, because of the excess newline, the comma ends up on the next line, which is ugly. This kwarg doesn't change the default, but it does allow us to turn that off.

Reply:
It was a comment in https://github.com/JuliaStats/Distributions.jl/pull/1363. In general, we should follow https://docs.julialang.org/en/v1/manual/types/#man-custom-pretty-printing-1 and hence in particular define such verbose `show` methods only for `::MIME"text/plain"` (as done in #1363). However, it seems the Julia docs also don't add a newline character at the end of the `::MIME"text/plain"` implementation, so it seems the comment in the PR was consistent with the current behaviour in Distributions but probably this should be changed.

Reply:
I can add `::MIME"text/plain"`. How would you recommend handling the newline issue?

Reply:
Maybe let's open an issue and fix it in a separate PR?

Reply:
According to ColPrac changes of `show` are never breaking so we are free to change it at any time: https://colprac.sciml.ai/#changes-that-are-not-considered-breaking